Picture liking functionality

kyleoge opened the pull request:
Adds buttons during the end story reveal that let players 'like' a picture and tallies each picture's like count.

kyleoge committed:
Liking pictures

Adds a button during the end story reveal that let's players 'like' a picture and tallies each picture's like count.

ItsAlxl commented on Client/game.js:
```diff
@@ -482,6 +484,36 @@ socket.on('take completed stories', function (stories, numStages, commStrokes =
+	let im = document.getElementById(imageId);
+	if (im.src.includes('like_off.png')) {
+		im.src = 'like_on.png';
+		document.getElementById(imageId.replace('like', 'counter')).style.color = 'green';
```
Rather than changing the style in JS, you should add a class in the CSS (eg .liked) and add/remove the class in JS. That keeps it more abstract and also makes it easier to modify the style later

ItsAlxl commented on Client/game.js:
```diff
@@ -466,6 +466,8 @@ socket.on('take completed stories', function (stories, numStages, commStrokes =
                 scrollHtml += s.owners[j - 1] + ' drew:<br>';
                 scrollHtml += '<img width="480" height="384" class="art" src="' + strokesToDataUrl(s.images[idx]) + '">';
+				scrollHtml += '<br> <img class="like-button" src="like_off.png" id="like ' + i + ' ' + j + '" onclick="likeImage(\'like ' + i + ' ' + j + '\')">';
+				scrollHtml += '<span class="like-counter" id="' + 'counter ' + i + " " + j + '">0</span> <br>';
```
Couldn't `id="' + 'counter ' +` just be `id="counter ' +` ?

kyleoge replied:
oh yeah lol

ItsAlxl reviewed (changes requested):
1. Use consistent tab styles. The project uses 4 spaces
2. There is a lot of potential for abuse in this system, eg desynchronized like counts or someone liking a photo several times. Who has liked which picture should be kept track of by the server to prevent these issues
3. Why not also allow people to like prompts? :(

This has made me realize there should probably be an additional class in GameComponents for StoryStage, which would contain stage type (i.e. drawing or caption), content, owner, who has liked it, etc., and the currently-existing Story class would be refactored to accommodate that.

kyleoge committed:
Server keeps track of likes

The server will now keep track of likes to ensure no one cheats. Also fixed some indentation stuff, allegedly.

ItsAlxl commented on Client/game.js:
```diff
@@ -488,26 +488,24 @@ function likeImage(imageId) {
 	} else if (im.src.includes('like_on.png')) {
 		im.src = 'like_off.png';
-		document.getElementById(imageId.replace('like', 'counter')).style.color = 'black';
+		document.getElementById(imageId.replace('like', 'counter')).className = 'like-counter-off';
```
Probably would be better to use Element.classList.remove() and Element.classList.add() ([here are the docs](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList)) in case we want to put more than one class on these elements.

In that vein, it'd make sense to change `.like-counter-off` to just `.like-counter`, give both elements the `.like-counter` class in the HTML, then add/remove the `.like-counter-on` class here.

ItsAlxl reviewed (changes requested):
This still has the potential for desync issues. Instead of sending two different signals from the server to the client for increment/decrement, just send an update signal with the number of likes it should have.

Also fix the tabs and remove the console.log() calls.

kyleoge committed:
Formatting changes to make things slightly less awful
